New Palette class (fix #978)

rec opened the pull request:
While this fixes the issue as written, the other half will be https://github.com/ManiacalLabs/BiblioPixelAnimations/issues/41 (after this is released).

adammhaile commented on bibliopixel/util/colors/palettes.py:
```diff
@@ -0,0 +1,40 @@
+DEFAULT = 'rainbow'
+
+
+def get(name='rainbow'):
```
Doc on this method?

rec replied:
Ah, I think I need a little more documentation in general, and this method in particular.  I'll write it up and then get back to you quite soon...

adammhaile commented on scripts/nearest-color:
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
```
What is this used for?

rec replied:
This isn't really very important - I wrote this when I was extracting the palettes and didn't want to entirely throw the code away.

(I didn't want to exactly steal palettes from other sites, so I took palettes that existed and converted them to the nearest named colors.)

I'd be very willing to drop this, but my plan was actually to remove `bp color` (because I'm wanting to do less in `bp`) and merge it into this.

adammhaile replied:
ok.... I just wanted to make sure it was clear what this script was for. Please at least add a usage note (include why you would use it). But fine with keeping it.

rec commented:
All right!  I added a lot more clarifying documentation and changed some naming slightly - you now get a color from a _position_ in a Palette.  (I realized that `index` wasn't quite right, because an index should be an integer, and this is expected to be a float...)

rec commented:
Done!  Waiting for the CI...
